climpred/prediction.py

```diff
@@ -83,28 +94,20 @@ def compute_perfect_model(ds,
     """
     supervector_dim = 'svd'
     comparison = get_comparison_function(comparison)
-    if comparison not in [_m2m, _m2c, _m2e, _e2c]:
-        raise ValueError('specify comparison argument')
-
+    _validate_PM_comparison(comparison)
     metric = get_metric_function(metric)
-    if metric in [_pearson_r, _rmse, _mse, _mae]:
-        forecast, reference = comparison(ds, supervector_dim)
-        res = metric(forecast, reference, dim=supervector_dim)
-    # perfect-model only metrics
-    elif metric in [_nmae, _nrmse, _nmse, _ppp, _uacc]:
-        res = metric(ds, control, comparison, running, reference_period)
-    else:
-        raise ValueError('specify metric argument')
+    _validate_PM_metric(metric)
+
+    forecast, reference = comparison(ds, supervector_dim)
+
+    res = metric(forecast,
+                 reference,
+                 dim=supervector_dim,
+                 comparison=comparison)
     return res
 
 
-check_xarray([0, 1])
-def compute_reference(ds,
-                      reference,
-                      metric='pearson_r',
-                      comparison='e2r',
-                      nlags=None,
```

Comment:
@aaronspring, forgot to ask. Why take this away? I like having the `nlags=None` keyword. By default, it takes the length of the lead dimension for `ds`. If one wants to compute less, they can designate the number of lags to compute.

Reply:
sure. the same way you can just input `ds.isel(lead=slice(0,5))`. its longer than `nlags=5`, sure, but for me its a questions of keeping the functions as slim as needed.

I prefer to have the nlags outside the core compute functions. I see them rather in the classes to subselect leads before inputing them into the core compute functions. Sounds OK to you? @bradyrx

Reply:
I'm still not convinced. I agree that slicing will do the same thing, but since it's a non-required keyword (`nlags`), it keeps things slim for the user. I would want to keep my full decadal prediction object, i.e., over the full time frame so I am not slicing it all the time. Then if I just quickly need to check a few lags I could run the `nlags` command.

We can keep this out for now but I might try to re-implement it in another PR.